### ISSUE TITLE
Overlapping masks predicted causing lower val mAP

### ISSUE THREAD

**Issue by davodogster** [dbolya/yolact#555]
How can I fix this strange thing the model is doing, overlapping DISJOINTED predictions shown in red.

![image](https://user-images.githubusercontent.com/46079516/96824362-188f8280-148b-11eb-8196-5b7caf6bac43.png)

I'm training with RLE, is crowd = 0 because is crowd = 1 doesn't train.

Regards, Sam

**Comment by areebsyed** [comment]
I think you have to do a post processing like doing something like detecting the contours of masks and then putting a condition that the two masks dont overlap! I want to do it too but will do it later. Let me know if you find some code solution

**Comment by davodogster** [comment]
Hi @areebsyed Haven't found a code solution yet sorry. Have you?

**Comment by amaanda** [comment]
Are using --no_crop = True?

https://github.com/dbolya/yolact/blob/1d3f7bb94f48dff6324b22483161374c35567306/eval.py#L96

**Comment by areebsyed** [comment]
> Hi @areebsyed Haven't found a code solution yet sorry. Have you?

@davodogster  no :/

**Comment by mmiisskk** [comment]
I would like to ask:
How big is your dataset and how long it takes, to get this results?

I usually train 12 hours and get just one detection in whole photo where is many same things as in yours. But you get really nice result.
Thanks.

**Comment by davodogster** [comment]
> Are using --no_crop = True?
> 
> https://github.com/dbolya/yolact/blob/1d3f7bb94f48dff6324b22483161374c35567306/eval.py#L96

No I never knew about that! what does it do ?

**Comment by davodogster** [comment]
@mmiisskk Hi Misky, 
How many classes do you have and how big (how may pixel width) are the objects after they are resized?

YOLACT pretrained with data augmentation img_size 1000 (default settings I think) is very powerful.
I just trained this on 40 images, val on 8. That is a testing image.

**Comment by mmiisskk** [comment]
@davodogster Hi,
thanks for reply.
My train dataset has 26 images and val dataset 10 images. Size of these images is 880x700 px. (objects size is 78x82 px)
After resize to 550x550 px for training/val set, objects size is 47x66 px.

**Comment by amaanda** [comment]
@davodogster it cutts off the parts of the masks that are not within the bounding box. Default is False, maybe you can test setting it True

```
parser.add_argument('--no_crop', default=False, dest='crop', action='store_false',
                        help='Do not crop output masks with the predicted bounding box.')

```

**Comment by davodogster** [comment]
@amaanda when I run eval.py with that flag --no_crop=True it says --no_crop: ignored explicit argument 'True'.

**Comment by PrajwalCogniac** [comment]
Hey @davodogster @mmiisskk 
Did you find any solution to this problem ?
